### ISSUE TITLE
Please provide instructions on how to set it up (and use) for Safari OS X / KyPass Companion

### ISSUE THREAD

**Issue by walkky** [mmichaa/passafari.safariextension#9]
I managed to install the plugin and set URL-Format and Submit-URL-Format to Full (per #2 ) and I've got a Passafari icon on the toolbar, but stuck on what to do next. Questions to start with:
1. How can I make sure it works?
2. How can I invoke auto-fill for the site existing in my password database?
3. Is there any way to copy password to clipboard, in order to paste it manually?

Would be helpful to find answers in the docs.

**Comment by arpost1** [comment]
Really not sure how to use the extension. Can't seem to insert passwords. Am I supposed to just click on it?

**Comment by mmichaa** [comment]
@esat7 Yeah, just click on it. I know, it doesn't provide much feedback. I'm willing to admit that is something that could be improved. On some websites the extension might not be sure, which fields are the login and password fields. You can help the extension by placing the cursor in one of them before the click on the extension's icon. I hope you get the extension working.

**Comment by mmichaa** [comment]
@esat7 …and I suppose that you already associated the extension with your KeePass application.

**Comment by arpost1** [comment]
I think I have associated it? When I clicked it, I get a prompt to name a key in KeePassXC. I then name it, click into the field on the website, and click the extension without results. Are there additional association steps?

I'm really excited about this extension by the way.

**Comment by arpost1** [comment]
So this is interesting... It works some of the time when I hit the extension, but it always prompts me to give the request a unique name. I thought it should only require that once and then be done?

**Comment by mmichaa** [comment]
I'm using KeePassXC as well. How do you set up the URL formats? I have both on _"Protocol & Host"_:

![image](https://user-images.githubusercontent.com/68866/33369188-0e6596f2-d4f4-11e7-8e7c-65317a0ae9ae.png)

The thing with the prompt _always_ popping up, seems to be a bug (related to #6). I have some kind of _workaround_: After I have associated it once, I click on _cancel_ the next time the prompt appears. Then on the next clicks on the extension icon, it works for me as usual until I reopen the KeePassXC the next time 😔 

I would really like to fix that, but there is currently not possible for me to release a new version of the extension. Because Apple cancelled their free(!) Safari Developer Program and it's now part of the $99/year Apple Developer Programm 😔  _(which I would need only for this extension)_

**Comment by arpost1** [comment]
That seems to have fixed it for me, actually! Very excited. I appreciate the work you've done on this already.

That is pretty crappy that they got rid of the free option. Not cool, Apple!
